### ISSUE TITLE
Cyclomatic complexity analysis and complex functions which need fixing

### ISSUE THREAD

**Issue by northwestwitch** [Clinical-Genomics/scout#2984]
I've just done a code complexity using [Radon](https://pypi.org/project/radon/). It's the same software also used by codefactor to find the complex functions that need fixing. The result of the analysis is quite good (which we knew), but I've opened this issue to highlight those that are still complex (D or worse). We could use them as simple small tasks to work on when we have little time available, perhaps starting not from the tests and from the F and E ones?

- [ ] tests/update/test_update_compounds.py
  - [ ]   F 14:0 test_compounds_region - D
- [ ] tests/parse/test_parse_clnsig.py
  - [ ]   F 7:0 test_parse_classic_clnsig - D
- [ ] tests/commands/view/test_view_cases.py
  - [ ]   F 11:0 test_view_cases - D
- [ ] tests/commands/view/test_view_cases_cmd.py
  - [ ]   F 7:0 test_view_cases - D
- [ ] tests/commands/export/test_export_variant_cmd.py
  - [ ]   F 7:0 test_export_variants - D
- [ ] tests/commands/export/test_export_cases_cmd.py
  - [ ]   F 7:0 test_export_cases - E
- [ ] tests/build/test_build_case.py
  - [ ]   F 9:0 test_build_case - D
- [ ] scout/parse/ensembl.py
  - [ ]   F 7:0 parse_ensembl_line - E
- [ ] scout/parse/case.py
  - [ ]   F 161:0 parse_case_data - D
- [ ] scout/parse/variant/frequency.py
  - [ ]   F 1:0 parse_frequencies - D
- [ ] scout/parse/variant/transcript.py
  - [ ]   F 10:0 parse_transcripts - F
- [ ] scout/parse/variant/genotype.py
  - [ ]   F 41:0 parse_genotype - F
- [ ] scout/parse/variant/clnsig.py
  - [ ]   F 6:0 parse_clnsig - D
- [ ] scout/parse/variant/variant.py
  - [ ]   F 25:0 parse_variant - F
- [ ] scout/server/blueprints/variants/controllers.py
  - [ ]   F 326:0 parse_variant - D
- [ ] scout/server/blueprints/cases/controllers.py
  - [ ]   F 56:0 case - F
- [ ] scout/server/blueprints/variant/controllers.py
  - [ ]   F 158:0 variant - E
- [ ] scout/server/blueprints/variant/verification_controllers.py
  - [ ]   F 21:0 variant_verification - D
- [ ] scout/adapter/mongo/query.py
  - [ ]   M 164:4 QueryHandler.build_query - E
  - [ ]   M 487:4 QueryHandler.secondary_query - D
- [ ] scout/adapter/mongo/variant_loader.py
  - [ ]   M 349:4 VariantLoader._load_variants - D
  - [ ]   M 579:4 VariantLoader.load_variants - D
- [ ] scout/adapter/mongo/case.py
  - [ ]   M 222:4 CaseHandler.cases - D
  - [ ]   M 150:4 CaseHandler._populate_name_query - D
- [ ] scout/adapter/mongo/variant.py
  - [ ]   M 832:4 VariantHandler.get_region_vcf - D
- [ ] scout/adapter/mongo/institute.py
  - [ ]   M 36:4 InstituteHandler.update_institute - D
- [ ] scout/load/setup.py
  - [ ]   F 41:0 setup_scout - D
- [ ] scout/commands/update/case.py
  - [ ]   F 67:0 case - D
- [ ] scout/build/case.py
  - [ ]   F 36:0 build_case - E
- [ ] scout/build/variant/variant.py
  - [ ]   F 9:0 build_variant - F
- [ ] scout/build/genes/hgnc_gene.py
  - [ ]   F 18:0 build_hgnc_gene - D
- [ ] scripts/check_gene_symbols_changed_in_panels.py
  - [ ]   F 49:0 check_panels - D
- [ ] scripts/update_files_path.py
  - [ ]   F 34:0 do_replace - E

**Comment by mikaell** [comment]
Fun! I've started with `scout/parse/variant/variant.py`.

**Comment by mikaell** [comment]
I've refactored a couple of modules under `scout/parse/` in #3076.

**Comment by mikaell** [comment]
Should refactoring PRs include only one module/function or several?

Maybe it's nicer, easier to review and easier to roll-back if we do only one module per PR. To the cost of slightly more overhead.

**Comment by northwestwitch** [comment]
> Maybe it's nicer, easier to review and easier to roll-back if we do only one module per PR. To the cost of slightly more overhead.

It's your choice I guess. One per PR is easier to review as well..

**Comment by dnil** [comment]
Time for an update on the list perhaps @northwestwitch? 😊

**Comment by northwestwitch** [comment]
> Time for an update on the list perhaps @northwestwitch? 😊

Yes, situation would have changed. I'll update a new analysis sometime soon
